packages/web/components/Quest/QuestFilter.tsx

```diff
@@ -34,89 +39,212 @@ export const QuestFilter: React.FC<Props> = ({
+  const [status, setStatus] = useState<ValueType>(statusOptions[0]);
+  const [guild, setGuild] = useState<ValueType>(guildOptions[0]);
+  const [roles, setRoles] = useState<ValueType[]>([]);
+
   return (
-    <Wrap justify="space-between">
-      <WrapItem>
-        <Wrap>
-          <WrapItem>
-            <MetaSelect
-              value={queryVariables.limit as number}
-              onChange={(e) =>
-                setQueryVariable('limit', Number(e.target.value))
+    <Wrap justifyContent="center">
+      <Wrap
+        transition="all 0.25s"
+        py="6"
+        style={{ backdropFilter: 'blur(7px)' }}
+        position="sticky"
+        top="-1px"
+        borderTop="1px solid transparent"
+        zIndex="1"
+        justifyContent="center"
+        w="100%"
+        maxW="79rem"
+        bg="whiteAlpha.200"
+        px="1.5rem"
+        borderRadius="6px"
+      >
+        <MetaFilterSelectSearch
+          title={`Limit: ${limit.label}`}
+          tagLabel=""
+          styles={metaFilterSelectStyles}
+          hasValue={false}
+          value={limit}
+          onChange={(value) => {
+            const values = value as ValueType[];
+            const v = values[values.length - 1];
+            if (v) {
+              setLimit(v);
+              setQueryVariable('limit', Number(v.value));
+            }
+          }}
+          options={limitOptions}
+          disableEmpty
+        />
+        <MetaFilterSelectSearch
+          title={`Order: ${order.label}`}
+          tagLabel=""
+          styles={metaFilterSelectStyles}
+          hasValue={false}
+          value={order}
+          onChange={(value) => {
+            const values = value as ValueType[];
+            const o = values[values.length - 1];
+            if (o) {
+              setOrder(o);
+              setQueryVariable('order', o.value);
+            }
+          }}
+          options={orderOptions}
+          disableEmpty
+        />
+        <MetaFilterSelectSearch
+          title={`Status: ${status.label}`}
+          tagLabel=""
+          styles={metaFilterSelectStyles}
+          hasValue={false}
+          value={status}
+          onChange={(value) => {
+            const values = value as ValueType[];
+            const s = values[values.length - 1];
+            if (s) {
+              setStatus(s);
+              setQueryVariable('status', s.value);
+            }
+          }}
+          options={statusOptions}
+          disableEmpty
+        />
+        {aggregates.guilds.length && (
+          <MetaFilterSelectSearch
+            title={`Guild: ${guild.label}`}
+            tagLabel=""
+            styles={metaFilterSelectStyles}
+            hasValue={false}
+            value={guild}
+            onChange={(value) => {
+              const values = value as ValueType[];
+              const g = values[values.length - 1];
+              if (g) {
+                setGuild(g);
+                setQueryVariable('guild_id', g.value);
```

Comment:
After you've done the camel cased custom names mentioned at the beginning of this review.

```suggestion
                setQueryVariable('guildId', g.value);
```